File: 2032-LargestOddNumberInString/2032-LargestOddNumberInString.py
# Last updated: 07/04/2026, 11:35:36
class Solution:
    def largestOddNumber(self, num: str) -> str:
        l = [1,3,5,7,9]
        j = len(num)-1
        while j >= 0 :
            if int(num[j])%2 == 1 :
                return num[:j+1]
            j-=1
    

        return ""
        

File: 2032-LargestOddNumberInString/2032-LargestOddNumberInString.cpp
// Last updated: 07/04/2026, 11:35:32
class Solution {
public:
    string largestOddNumber(string num) {
        
        int n = num.size();
        string ans; 
        int m = -1;

        for ( int i= n-1;i>=0;i--){
            int v = num[i]-'0';
            if ( v%2 == 1){
                m = i;
                break;
               
            }
        }
        if ( m == -1) return ans;


        for ( int i = 0;i<=m;i++){
            ans+=num[i];
        }
        return ans ;




    }
};